source/loop.cpp

```diff
@@ -350,6 +409,64 @@ double Loop::CalculateRadiativeLoss(double temperature)
 	return chi * std::pow( 10.0, (alpha*log_temperature) );
 }
 
+double Loop::CalculateRadiativeLoss(double temperature, double density)
+{
+    double abundance_array[] = {1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0};  // Discrete values available for abundance factors
```

Comment:
Hardcoding these values makes me really nervous. My preferred solution would be that they are stored in the file(s) and then read in from there. Otherwise, we run the risk of the files and these hardcoded values being out of sync. It is also makes it easier to add new abundance factors if we do it this way.